app/models/gemini_client.py

```
@@ -76,27 +77,87 @@ def ask(self, prompt: str):
+                                    f"GeminiService: JSONパースに失敗しました。候補文字列: {candidate[:500]}",
+                                    flush=True,
+                                )
+                                print(f"JSONDecodeError: {e}", flush=True)
+                                response_dict = None
 
             # リスト形式の場合は最初の要素を取得
             if isinstance(response_dict, list) and len(response_dict) > 0:
                 response_dict = response_dict[0]
 
+            if response_dict is None:
+                raise ValueError("Could not parse Gemini response as JSON")
```

Ruff (0.14.3)

Abstract `raise` to an inner function

(TRY301)